### PR TITLE
wc: tx simulation

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-####

### CODE REVIEW

File: src/components/transaction/TransactionMessage.js [hunk 1]
```diff
@@ -14,17 +14,12 @@ const deviceWidth = deviceUtils.dimensions.width;
 const horizontalPadding = 24;
 
 const Container = styled(Row)({
-  maxHeight: ({ maxHeight }) => maxHeight,
```
Comment (skylarbarrera) [inline src/components/transaction/TransactionMessage.js:17]: this file is deprecated and has been moved into the signTransactionSheet

will open a PR post merge to clean up old components

File: src/graphql/config.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ exports.config = {
     document: './queries/metadata.graphql',
     schema: { method: 'GET', url: 'https://metadata.p.rainbow.me/v1/graph' },
   },
+  simulation: {
```
Comment (skylarbarrera) [inline src/graphql/config.js:15]: simulation needs POST requests this should only be used for simulation as POSTs mess with the cloudflare CDN

File: src/model/wallet.ts [hunk 3]
```diff
@@ -90,7 +91,7 @@ interface WalletInitialized {
 
 interface TransactionRequestParam {
   transaction: TransactionRequest;
-  existingWallet?: Wallet;
+  existingWallet?: Signer;
```
Comment (skylarbarrera) [inline src/model/wallet.ts:94]: our `LedgerSigner` class is the reason for this change, `Wallet` is a implementation of `Signer`

File: src/navigation/Routes.android.tsx [hunk 4]
```diff
@@ -382,10 +383,14 @@ function BSNavigator() {
       />
       <BSStack.Screen name={Routes.MINTS_SHEET} component={MintsSheet} />
       <BSStack.Screen
-        component={TransactionConfirmationScreen}
+        component={SignTransactionSheet}
```
Comment (skylarbarrera) [inline src/navigation/Routes.android.tsx:386]: keeping this as a 1 to 1 replacement for the existing flow

File: src/navigation/Routes.ios.tsx [hunk 5]
```diff
@@ -326,10 +328,11 @@ function NativeStackNavigator() {
         {...restoreSheetConfig}
       />
       <NativeStack.Screen
-        component={TransactionConfirmationScreen}
+        component={SignTransactionSheet}
         name={Routes.CONFIRM_REQUEST}
         options={{
-          allowsDragToDismiss: true,
+          allowsDragToDismiss: false,
```
Comment (skylarbarrera) [inline src/navigation/Routes.ios.tsx:334]: we are no longer allowing dismissal of WC requests as a pending request will block any new incoming requests

File: src/navigation/bottom-sheet/types.d.ts [hunk 6]
```diff
@@ -122,6 +123,7 @@ export type BottomSheetNavigationOptions = Partial<
 > & {
   backdropOpacity?: number;
   backdropColor?: string;
+  backdropPressBehavior?: BackdropPressBehavior;
```
Comment (skylarbarrera) [inline src/navigation/bottom-sheet/types.d.ts:126]: same thing here but for android so we can block dismissal

File: src/screens/SignTransactionSheet.tsx [hunk 7]
```diff
@@ -0,0 +1,2242 @@
...
+    transactionDetails?.requestId,
+    transactionDetails?.walletConnectV2RequestValues,
+    transactionDetails?.peerId,
+    currentNetwork,
+    accountInfo.address,
+    accountInfo.isHardwareWallet,
+    callback,
+    closeScreen,
+    dispatch,
+    onCancel,
```
Comment (dereknelson) [inline src/screens/SignTransactionSheet.tsx:760]: W dependency array

File: src/screens/SignTransactionSheet.tsx [hunk 8]
```diff
@@ -0,0 +1,2242 @@
...
+                  >
+                    <Text
+                      color="label"
+                      numberOfLines={1}
+                      size="20pt"
+                      weight="heavy"
+                    >
+                      {transactionDetails.dappName}
+                    </Text>
+                    {false && <VerifiedBadge />}
```
Comment (dereknelson) [inline src/screens/SignTransactionSheet.tsx:1081]: 🤔

**Review by dereknelson — APPROVED** [review]
2,242 lines of SignTransactionSheet later 😮‍💨 i'm thinkin this is good to go

File: src/screens/SignTransactionSheet.tsx (hunk 8 above)
Comment (skylarbarrera) [reply src/screens/SignTransactionSheet.tsx:1098]: yeah we can't verify urls via WC consistently but leaving there cause there may be path forward soon

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
https://linear.app/rainbow/issue/APP-921/skylarsimulation-bug-epic

up to commit f474a318500ebcbecfa4a28f263f8a3094594b40 just two issues
latest commit ebb20712c99417ad473aa21a72676ff0a139252a, many gas issues 

all explained in above bug epic

File: src/screens/SignTransactionSheet.tsx [hunk 9]
```diff
@@ -0,0 +1,2300 @@
...
+              amount={change?.quantityAllowed || '10'}
+              asset={change?.asset}
+              eventType="approve"
+            />
+          );
+        })}
+        {simulation?.out?.map(change => {
+          return (
+            <SimulatedEventRow
+              key={`${change?.asset?.assetCode}-${change?.quantity}`}
```
Comment (greg-schrammel) [inline src/screens/SignTransactionSheet.tsx:1378]: yo I know this is merged already was curious how we implemented the same thing lol

and not that it matters a lot but I think this key is gonna conflict in a case like opensea pro buying lets say rainbow zorbs in bulk, same asset code and quantity, but from multiple sellers

idk how to do it either I added the index to the key even tho it's not good practice, considering this array won't change it's probably safe (unless we refetch and even then its hard to hit)

File: src/screens/SignTransactionSheet.tsx [hunk 10]
```diff
@@ -0,0 +1,2300 @@
...
+              amount={change?.quantity || '10'}
```
Comment (greg-schrammel) [inline src/screens/SignTransactionSheet.tsx:1379]: is this `|| '10'` desired?

**Pull request merged**